### PR TITLE
Path Planner Update

### PR DESCRIPTION
**Pull request by red-eight**
Corrects const-correctness of some of `Path`'s member functions and implements a generic search that turns BSF and DFS into one-liners.

### CODE REVIEW

File: igvc/src/pathplanner/GraphSearch.hpp [hunk 1]
```diff
@@ -35,16 +35,49 @@ class PathComparator
...
+	// Runs a generic search that DFS and BSF can call.
+	// Getting this to work with the current implementation of A* is possible, but difficult.
+	template <class StateType, class ActionType, template <typename ...> class FrontierType>
+	static auto GenericSearch(SearchProblem<StateType, ActionType> &problem) -> Path<StateType, ActionType>
+	{
+		using frontier_traits = FrontierTraits<StateType, ActionType, FrontierType>;
+		using frontier_type = typename frontier_traits::frontier_type;
+        
+		set<StateType> expanded;
+        frontier_type frontier;
```
Comment (rmkeezer) [inline igvc/src/pathplanner/GraphSearch.hpp]: What is going on with the spacing here?

File: igvc/src/pathplanner/GraphSearch.hpp [hunk 2]
```diff
@@ -54,24 +87,26 @@ class GraphSearch
...
+                expanded.insert(lastState);
+
+                if(problem.isGoal(lastState))
                 {
                     return path;
                 }
-                list<ActionType> legalActions = problem.getActions(path.getLastState());
-                StateType last = path.getLastState();
+                
+				auto legalActions = problem.getActions(lastState);
```
Comment (rmkeezer) [inline igvc/src/pathplanner/GraphSearch.hpp]: Indention is pretty messed up in this file

File: igvc/src/pathplanner/GraphSearch.hpp (hunk 1 above)
Comment (red-eight) [inline igvc/src/pathplanner/GraphSearch.hpp]: Sorry about that; I'm not sure what happened. 
I'll try to fix the spacing tomorrow.
Should I do another pull request once it's done?

Comment (jgkamat) [inline igvc/src/pathplanner/GraphSearch.hpp]: It seems like your spaces switched to tabs somehow. You could do a find replace on tabs to 4 spaces, or just re-indent the entire file once you've changed the editor settings.

No need for another PR, just push more commits to this branch!

Comment (jgkamat) [inline igvc/src/pathplanner/GraphSearch.hpp]: The github editor also defaults to tabs, so your new commit also has 'broken' formatting.

Let me know if you need help with this, I can help you out and make the changes on my end.

Comment (red-eight) [inline igvc/src/pathplanner/GraphSearch.hpp]: If you don't mind, then go for it.
If you don't have time, then I will fix this tomorrow.

**Comment by jgkamat** [comment]
@red-eight this latest commit should have fixed your problem. @rmkeezer needs to fix windows line endings on graphsearch, thats why the whole file changed.

**Comment by rmkeezer** [comment]
Looks good + editorconfig was definitely needed!

**Pull request merged**